Sources/web3swift/Web3/Web3+GasOracle.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  Web3+GasOracle.swift
+//  web3swift
+//
+//  Created by Yaroslav on 31.03.2022.
+//  Copyright © 2022 web3swift. All rights reserved.
+//
+
+import Foundation
+import BigInt
+
+extension Web3 {
+    /// Oracle is the class to of a transaction fee suggestion
+    ///
+    /// It designed for EIP-1559 transactions only.
+    final public class Oracle {
+        private var latestBlock: Block?
+
+        /// Web3 provider by wich accessing to the blockchain
+        private let web3Provider: web3
+
+        /// Ethereum scope shortcut
+        private var eth: web3.Eth { web3Provider.eth }
+
+        /// Number of block to caltulate statistics
+        public private(set) var blocksNumber: BigUInt
+
+        /// Number of transacrtions to filter block for tip calculation
+        public private(set) var transactionsNumber: BigUInt
+
+        /// Oracle initializer
+        /// - Parameters:
+        ///   - provider: Web3 Ethereum provider
+        ///   - blocksNumber: Number of block to caltulate statistics
+        ///   - transactionsNumber: Number of transacrtions to filter block for tip calculation
+        public init(_ provider: web3, blocksNumber: BigUInt = 20, transactionsNumber: BigUInt = 50) {
+            self.web3Provider = provider
+            self.blocksNumber = blocksNumber
+            self.transactionsNumber = transactionsNumber
+        }
+
+        private func calcBaseFee(for block: Block?) -> BigUInt {
+            guard let block = block else { return 0 }
+            return Web3.calcBaseFee(block)
+        }
+
+        private func calculateStatistic(_ data: [BigUInt], _ statistic: Statistic) throws -> BigUInt {
+            let sortedData = data.sorted()
+            let noAnomalyArray = sortedData.cropAnomalyValues()
+
+            guard !noAnomalyArray.isEmpty else { throw Web3Error.unknownError }
+
+            switch statistic {
+            // Force unwrapping is ok, since array checked for epmtiness above
+            case .minimum: return noAnomalyArray.min()!
```

Comment:
we should not use force unwrap

Reply:
Please pay attention that at line 51 its explicitly checks that array is not empty, therefore it's acceptable to use forse unwrap, since each function beyond returns nil only if arrays is empty.

Reply:
I agree that the forced unwrap is fine here

Add the switlint comment
`// swiftlint:disable force_unwrapping` before the block
and then `// swiftlint:enable force_unwrapping` after the block so that yo don't get the warnings